pyproj/geod.py

```diff
@@ -336,17 +353,333 @@ def npts(
+        '46.299 -89.279'
+        '46.830 -94.156'
+        '47.149 -99.112'
+        '47.251 -104.106'
+        '47.136 -109.100'
+        '46.805 -114.051'
+        '46.262 -118.924'
+        >>> # test with radians=True (inputs/outputs in radians, not degrees)
+        >>> import math
+        >>> dg2rad = math.radians(1.)
+        >>> rad2dg = math.degrees(1.)
+        >>> r = g.inv_intermediate(
+        ...    dg2rad*boston_lon,
+        ...    dg2rad*boston_lat,
+        ...    dg2rad*portland_lon,
+        ...    dg2rad*portland_lat,
+        ...    10,
+        ...    radians=True
+        ... )
+        >>> for lon,lat in zip(r.lons, r.lats): f'{rad2dg*lat:.3f} {rad2dg*lon:.3f}'
+        '43.528 -75.414'
+        '44.637 -79.883'
+        '45.565 -84.512'
+        '46.299 -89.279'
+        '46.830 -94.156'
+        '47.149 -99.112'
+        '47.251 -104.106'
+        '47.136 -109.100'
+        '46.805 -114.051'
+        '46.262 -118.924'
+
+        Parameters
+        ----------
+        lon1: float
+            Longitude of the initial point
+        lat1: float
+            Latitude of the initial point
+        lon2: float
+            Longitude of the terminus point
+        lat2: float
+            Latitude of the terminus point
+        npts: int, optional
+            Number of points to be returned
+            npts == 0 iff del_s != 0
+        del_s: float, optional
+            delimiter distance between two successive points
+            del_s == 0 iff npts != 0
+        radians: bool, optional
+            If True, the input data is assumed to be in radians.
+        initial_idx: int, optional
+            if initial_idx==0 then the initial point would be included in the output
+            (as the first point)
+        terminus_idx: int, optional
+            if terminus_idx==0 then the terminus point would be included in the output
+            (as the last point)
+        flags: GeodIntermediateFlag, optional
+            1st: round/ceil/trunc (see GeodIntermediateFlag.NPTS_*)
+            2nd: update del_s to the new npts or not (see GeodIntermediateFlag.DEL_S_*)
+            3rd: iff out_azis=None, indicates if to save or discard the azimuths
+                 (see GeodIntermediateFlag.AZIS_*)
+            default - round npts, update del_s accordingly, discard azis
+        out_lons: array, :class:`numpy.ndarray`, optional
+            Longitude(s) of the intermediate point(s)
+            If None then buffers would be allocated internnaly
+        out_lats: array, :class:`numpy.ndarray`, optional
+            Latitudes(s) of the intermediate point(s)
+            If None then buffers would be allocated internnaly
+        out_azis: array, :class:`numpy.ndarray`, optional
+            az12(s) of the intermediate point(s)
+            If None then buffers would be allocated internnaly
+            unless requested otherwise by the flags
+
+        Returns
+        -------
+        GeodIntermediateReturn: NamedTuple
```

Comment:
Instead of adding the definition on each return, thoughts about adding `__doc__` to the `GeodIntermediateReturn` as it is done [here](https://github.com/pyproj4/pyproj/blob/2d12d5aa719ea989c8f6abdb81916d43200fe9c3/pyproj/database.pyx#L128-L149) and then adding a reference to `GeodIntermediateReturn` in the docs in `docs/api/geod.rst`?

Reply:
OK. 
Should I move `GeodIntermediateReturn` to someplace else (new `pyproj.types.py`?)
Otherwise how could I declare it correctly in `_geod.pyi` ? I've tried `from pyproj.geod import GeodIntermediateReturn` before but it fails, and as it is now I get incorrect type hints from the IDE (as it sees it as `NamedTuple` as I decalred in `_geod.pyi`.

Reply:
No need to move it to a new place. Here an example of declaring the tuple type: [ref](https://github.com/pyproj4/pyproj/blob/2d12d5aa719ea989c8f6abdb81916d43200fe9c3/pyproj/database.pyi#L25).

Reply:
nice.